src/LiquidityPool.sol

```diff
@@ -215,23 +210,36 @@ contract LiquidityPool is Auth, IERC4626 {
     /// @notice Request asset deposit for a receiver to be included in the next epoch execution.
     /// @notice Request can only be called by the owner of the assets
     ///         Asset is locked in the escrow on request submission
-    function requestDeposit(uint256 assets) public {
-        manager.requestDeposit(address(this), assets, msg.sender);
-        emit DepositRequest(msg.sender, assets);
+    function requestDeposit(uint256 assets, address operator) public {
+        require(IERC20(asset).balanceOf(msg.sender) >= assets, "LiquidityPool/insufficient-balance");
+        require(
+            manager.requestDeposit(address(this), assets, msg.sender, operator), "LiquidityPool/request-deposit-failed"
+        );
+        SafeTransferLib.safeTransferFrom(asset, msg.sender, address(escrow), assets);
+        emit DepositRequest(msg.sender, operator, assets);
     }
 
     /// @notice Similar to requestDeposit, but with a permit option
-    function requestDepositWithPermit(uint256 assets, address owner, uint256 deadline, uint8 v, bytes32 r, bytes32 s)
-        public
-    {
-        _withPermit(asset, owner, address(manager), assets, deadline, v, r, s);
-        manager.requestDeposit(address(this), assets, owner);
-        emit DepositRequest(owner, assets);
+    function requestDepositWithPermit(
+        uint256 assets,
+        address operator,
+        address owner,
+        uint256 deadline,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) public {
+        _withPermit(asset, owner, address(this), assets, deadline, v, r, s);
+        require(manager.requestDeposit(address(this), assets, owner, operator), "LiquidityPool/request-deposit-failed");
```

Comment:
I don't think that we can pass the operator here.
Malicious operator that is also on the restricted token allowlist could frontrun the permit and pass their own operator address to the function